tidb-binlog/pump_client/pump.go

```diff
@@ -73,12 +75,24 @@ type PumpStatus struct {
 // NewPumpStatus returns a new PumpStatus according to node's status.
 func NewPumpStatus(status *node.Status, security *tls.Config) *PumpStatus {
 	pumpStatus := PumpStatus{
-		Status:   *status,
-		security: security,
+		Status:         *status,
+		security:       security,
+		reCreateClient: true,
 	}
 	return &pumpStatus
 }
 
+func (p *PumpStatus) markReCreateClient() {
+	p.reCreateClient = true
```

Comment:
need lock?

Reply:
has hold the lock  in `updatePump` only this func will call this

Reply:
it's better to do it, cuz we don't hold lock in other functions who calls `WriteBinlog`, same with `close`